carma_wm_ctrl/include/carma_wm_ctrl/WMBroadcaster.h

```diff
@@ -53,26 +56,40 @@ class WMBroadcaster
    * \brief Callback to set the base map when it has been loaded
    *
    * \param map_msg The map message to use as the base map
-   *
    */
   void baseMapCallback(const autoware_lanelet2_msgs::MapBinConstPtr& map_msg);
 
+  /*!
+   * \brief Callback to set the base map georeference (proj string)
+   *
+   * \param georef_msg Proj string that specifies the georeference of the map. 
+   * It is used for transfering frames between that of geofence and that of the vehicle
+   */
+  void geoReferenceCallback(const std_msgs::String& geo_ref);
+
   /*!
    * \brief Callback to add a geofence to the map
    *
-   * \param geofence_msg The geofence to add. TODO replace with actual message type once defined
+   * \param geofence_msg The ros msg of the geofence to add. 
    */
-  void geofenceCallback(const Geofence& gf);
+  void geofenceCallback(const cav_msgs::ControlMessage& geofence_msg);
 
   /*!
    * \brief Adds a geofence to the current map
    */
-  void addGeofence(const Geofence& gf);
+  void addGeofence(Geofence& gf);
 
   /*!
    * \brief Removes a geofence from the current map
    */
-  void removeGeofence(const Geofence& gf);
+  void removeGeofence(Geofence& gf);
+
+  /*!
+   * \brief Gets the affected lanelet or areas based on the geofence_msg
+   * NOTE: Currently only checks lanelets
+   * TODO add throws and whatnot
+   */
+  lanelet::Lanelets getAffectedLaneletOrAreas(const cav_msgs::ControlMessage& geofence_msg);
```

Comment:
If you intend to have this function work with areas as well then you should use the lanelet::LaneletOrArea data type as the return value.